website_sale_product_minimal_price/models/product_template.py

```diff
@@ -12,12 +14,31 @@ class ProductTemplate(models.Model):
     )
 
     def _compute_has_distinct_variant_price(self):
+        partner = self.env.context.get('partner')
+        pricelist_id = self.env.context.get('pricelist')
         for template in self:
             if template.product_variant_count > 1:
-                prices = template.product_variant_ids.mapped('website_price')
-                if len(prices) > 1:
+                prices = template._get_variants_price(pricelist_id, 1, partner)
+                if len(set(prices.values())) > 1:
                     template.has_distinct_variant_price = True
 
+    # TODO: Use ormcache but we need invalidate cache in all models involved
+    #  in a price change
+    @ormcache('self.id', 'pricelist_id', 'qty', 'partner')
+    def _get_variants_price(self, pricelist_id, qty, partner):
```

Comment:
Just in case you decide to go for the `ormcache_context` option:

```suggestion
    def _get_variants_price(self, qty):
```